app/controllers/admins_controller.rb

```diff
@@ -3,7 +3,50 @@ class AdminsController < ApplicationController
   before_filter :authenticate_admin!, only: [:index]
 
   def index
+  end
+
+  def new
+    @admin = Admin.new
+  end
+
+  def create
+    @company = Company.where(id: current_admin.company.id).first
+    @admin = @company.admins.build(admin_params)
+    if @admin.save_without_confirmation
+      redirect_to admins_path, notice: 'New agent added'
+    else
+      render :new
+    end
+  end
+
+  def edit
+    @admin = current_admin
```

Comment:
@admin not needed. current_admin will serve the purpose. it is also available in views